### DISCUSSION TITLE
Maxwell 2D: edit_external_circuit

### DISCUSSION THREAD

**Discussion by rickygill00** [ansys/pyaedt#4852] [Q&A] [answered]
Hello, 

When using edit_external_circuit I get the following error message in Maxwell 2D: "Invalid source name or source value" even though using this method returns True. 

Manually importing the .sph file works so not sure what the problem is.

**Comment by gmalinve** [comment]
Hi @rickygill00,

Could you please reply with a code snippet that could help me identify your issue? 
Also which AEDT version are you using?

Thanks,

Regards,

Giulia

**Comment by jvela018** [comment, marked as answer]
@rickygill00 ,

Have you been able to solve this issue? Without a code snippet there's not much we can help you with, however here are my two cents.

Note that the edit_external_circuit assumes that you've created a circuit netlist. In other words, you're either pointing to the already generated .sph file, or you need to generate one using MaxwellCircuit() and then provide edit_external_circuit with the required information.

If you follow the external circuit example you'll see that the creation of circuits is split into two main parts: 
- The one using the MaxwellCircuit instance (creation of the circuit returns MaxwellCircuit object): https://examples.aedt.docs.pyansys.com/version/dev/examples/low_frequency/general/external_circuit.html#Create-external-circuit

- The one using the Maxwell2d instance to update the circuit connected to the model windings: https://examples.aedt.docs.pyansys.com/version/dev/examples/low_frequency/general/external_circuit.html#Export-and-import-netlist

I hope this helps.
